.github/workflows/ci-core.yml

```diff
@@ -49,3 +49,10 @@ jobs:
         export PYTHONPATH=$(pwd)
         lit -v tests/filecheck/
         lit -v docs/Toy/examples/
+    
+    - name: Execute book tests
+      run: |
+        for f in docs/book/**/*.py; do
+          echo "Running book code: '$f'";
+          python "$f";
+        done
```

Comment:
can you please add this to make tests? this script should probably be running those tests also, but that's for another PR. Ideally `test book`.

Reply:
I am not sure if this script is the best way to do this in the first place. With your Python build system experience, would you do it differently?

Reply:
Actually why this format and not the way we have it in Toy, as just an extra project with source files and pytest tests?

Reply:
Could you detail a bit more? The files that will be added will most likely not have the usual "list of functions" format used with pytest.

Reply:
This is partially why I suggested moving the Toy tutorial. Even though it's not what we want to have in the book in the long run, it contains realistic snippets of code, and would immediately surface missing functionality.